src/index.js

```diff
@@ -184,7 +195,19 @@ exports.createFromBytes = (buf) => {
 }
 
 exports.createFromB58String = (str) => {
-  return new PeerIdWithIs(mh.fromB58String(str))
+  return exports.createFromCID(str) // B58String is CIDv0
+}
+
+exports.createFromCID = (cid) => {
+  if (typeof cid === 'string' || Buffer.isBuffer(cid)) {
+    cid = new CID(cid)
+  } else if (CID.isCID(cid)) {
+    CID.validateCID(cid) // throws on error
+  } else {
+    // provide more meaningful error than the one in CID.validateCID
+    throw new Error('Supplied cid value is neither String|CID|Buffer')
+  }
```

Comment:
TODO: validate the codec is `libp2p-key` or that it's a v0?

Reply:
This is tricky. 
We want people to be able to convert CIDv0→CIDv1 and be able to use it.

See scenario below:

- Execute `ipfs id`
  - You will see PeerID as Base58 (CIDv0) which has [implicit multicodec `dag-pb`](https://github.com/multiformats/cid#cidv0)
- How to convert PeerID to CIDv1?
  - Probably via `ipfs cid base32` or https://cid.ipfs.io util
- Caveat: conversion produces a valid CIDv1 with (now explicit) `dag-pb` multicodec
  - the multihash inside of CIDv1 still represents a valid PeerID, and this library does not care about the rest od CID anyway

I think we should check for both `libp2p-key` and `dag-pb`. 
Will update the PR to do just that + tests.